bin/main.rb

```diff
@@ -1,3 +1,60 @@
 #!/usr/bin/env ruby
 
-puts 'Hello world!'
+def print_board(arr)
+  puts "#{arr[0]} | #{arr[1]} | #{arr[2]}"
+  puts '..|...|..'
+  puts "#{arr[3]} | #{arr[4]} | #{arr[5]}"
+  puts '..|...|..'
+  puts "#{arr[6]} | #{arr[7]} | #{arr[8]}"
+end
+# Game begins
+puts 'Welcome to tic-tac-toe game...'
+puts 'Please choose one of the following options: Select 1 to start a new game, Select 2 to quit'
+option = gets.chomp
+
+# Start New Game
+if option.to_i == 1
+
+  # player input
+  puts 'Please enter the name of player one'
+  player_one = gets.chomp
+
+  puts 'Please enter the name of player two'
+  player_two = gets.chomp
+
+  # start Game
+  puts 'The game has begin'
+  game_not_finished = true
+
+  while game_not_finished
+
+    player_current = player_one
+    available_moves = [1, 2, 3, 4, 5, 6, 7, 8, 9]
+
+    # Players turn instruction
+    puts "#{player_current} It is your turn."
+    print_board(available_moves)
+    puts "Please input your move. You have these moves available: #{available_moves}"
+    current_player_selected_move = gets.chomp
+    available_moves[current_player_selected_move.to_i - 1] = 'X'
+    game_invalid_move = true
```

Comment:
This should be a conditional
game_invalid_move is only true if the preceding line is true

Reply:
Also, have it in mind that moves outside the range of 1-9 are also invalid.